images/ami_kms_key.tf

```diff
@@ -119,7 +119,7 @@ data "aws_iam_policy_document" "ami_kms_doc" {
       values = [
         for account in data.aws_organizations_organization.cool.accounts :
         "arn:aws:iam::${account.id}:role/ProvisionAccount"
-        if length(regexall("^env|^Playground Legacy$|^Shared Services$", account.name)) > 0
+        if length(regexall("^env.*\\(${local.this_account_type}\\)$|^Playground Legacy \\(${local.this_account_type}\\)$|^Shared Services \\(${local.this_account_type}\\)$", account.name)) > 0
```

Comment:
I think `^env[0-9]*\\(${local.this_account_type}\\)$` would be slightly less inclusive and thus slightly more correct.  Just in case we have a goofily-named account as some point (like `envaders (Production)`) that shouldn't get access to these AMIs.

Reply:
Good call- that makes sense to me, though I think your regex is missing a space (to handle the space between the last digit and the opening parenthesis), so we would end up with this:
`^env[0-9]* \\(${local.this_account_type}\\)$`

Reply:
Changed in https://github.com/cisagov/cool-accounts/pull/42/commits/da99eb22145d681a7f4a2878f67ea4cd7f16ef41.